modules/apps/export-import/export-import-service/src/main/java/com/liferay/exportimport/internal/scheduler/DeleteExpiredBackgroundTasksSchedulerJobConfiguration.java

```diff
@@ -0,0 +1,200 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.exportimport.internal.scheduler;
+
+import com.liferay.exportimport.configuration.ExportImportServiceConfiguration;
+import com.liferay.exportimport.configuration.ExportImportSystemConfiguration;
+import com.liferay.exportimport.kernel.background.task.BackgroundTaskExecutorNames;
+import com.liferay.petra.function.UnsafeRunnable;
+import com.liferay.portal.background.task.model.BackgroundTask;
+import com.liferay.portal.background.task.service.BackgroundTaskLocalService;
+import com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil;
+import com.liferay.portal.kernel.backgroundtask.constants.BackgroundTaskConstants;
+import com.liferay.portal.kernel.dao.orm.ActionableDynamicQuery;
+import com.liferay.portal.kernel.dao.orm.Property;
+import com.liferay.portal.kernel.dao.orm.PropertyFactoryUtil;
+import com.liferay.portal.kernel.dao.orm.RestrictionsFactoryUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.module.configuration.ConfigurationException;
+import com.liferay.portal.kernel.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.scheduler.SchedulerJobConfiguration;
+import com.liferay.portal.kernel.scheduler.TimeUnit;
+import com.liferay.portal.kernel.scheduler.TriggerConfiguration;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.util.Time;
+
+import java.util.Date;
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author To Trinh
+ * @author Ha Tang
+ */
+@Component(
+	configurationPid = {
+		"com.liferay.exportimport.configuration.ExportImportServiceConfiguration",
+		"com.liferay.exportimport.configuration.ExportImportSystemConfiguration"
+	},
+	service = SchedulerJobConfiguration.class
+)
+public class DeleteExpiredBackgroundTasksSchedulerJobConfiguration
+	implements SchedulerJobConfiguration {
+
+	@Override
+	public UnsafeRunnable<Exception> getJobExecutorUnsafeRunnable() {
+		return () -> _companyLocalService.forEachCompanyId(
+			companyId -> _deleteExpiredBackGroundTasks(companyId));
+	}
+
+	@Override
+	public TriggerConfiguration getTriggerConfiguration() {
+		return TriggerConfiguration.createTriggerConfiguration(
```

Comment:
If the `cleanupJobInterval()` value is 0, then we get an exception at portal startup:
https://github.com/liferay/liferay-portal/blob/fa01b638b5bd41ad7057f3fb274a710c99444a3a/portal-kernel/src/com/liferay/portal/kernel/scheduler/TriggerConfiguration.java#L47-L50
I think this should be avoided somehow. I haven't found any examples yet, how can we disable scheduled jobs in this case.
If it's not possible, we can add a non-zero default value to the config, or we can even hardcode this value.
The instance-level configuration still disables the actual deletion even if this job runs regularly.

Reply:
yeah, i found a sample in another app that catch the exception, let me see if that helps. thanks,

Reply:
> If the `cleanupJobInterval()` value is 0, then we get an exception at portal startup: https://github.com/liferay/liferay-portal/blob/fa01b638b5bd41ad7057f3fb274a710c99444a3a/portal-kernel/src/com/liferay/portal/kernel/scheduler/TriggerConfiguration.java#L47-L50 I think this should be avoided somehow. I haven't found any examples yet, how can we disable scheduled jobs in this case. If it's not possible, we can add a non-zero default value to the config, or we can even hardcode this value. The instance-level configuration still disables the actual deletion even if this job runs regularly.

Hi @vendeltoreki 
I wasn't able to disable the scheduled job when interval is 0. My workaround is hardcoding a far future scheduled date Dec 31 2099. Can you help check if that makes sense?
Thanks,